grpc-native/src/main/java/org/ballerinalang/net/grpc/nativeimpl/client/FunctionUtils.java

```diff
@@ -239,7 +234,7 @@ public static Object externBlockingExecute(Environment env, BObject clientEndpoi
             Message requestMsg = new Message(methodDescriptor.getInputType().getName(), payloadBValue);
             // Update request headers when request headers exists in the context.
             HttpHeaders headers = null;
-            if (headerValues != null && (TypeChecker.getType(headerValues).getTag() == TypeTags.OBJECT_TYPE_TAG)) {
+            if (headerValues instanceof BObject) {
```

Comment:
Shouldn't we need this null check?

Reply:
We don't need a null check. `instanceof` doesn't satisfy for null values.